### PR TITLE
Fixed #242025 - Added nonce

### PR DESCRIPTION
**Pull request by irrationalRock**
This PR addresses the following issue: https://github.com/Microsoft/TypeScript/issues/24205.

It adds `nonce` to both `HTMLScriptElement` and `HTMLStyleElement`.

Thanks for considering this PR.

### CODE REVIEW

**Comment by HolgerJeromin** [comment]
Shouldn't this change go to addingTypes.json?

**Comment by mhegazy** [comment]
@saschanaz should we add https://html.spec.whatwg.org/multipage/scripting.html#the-script-element instead?

**Comment by saschanaz** [comment]
@mhegazy #496 already did that. `nonce` is part of https://html.spec.whatwg.org/multipage/dom.html, but we probably can't add this right now because `GlobalEventHandlers` mixin should be processed first. Otherwise we'll lose lots of event handlers.

**Comment by saschanaz** [comment]
Also, `nonce` should be added to HTMLElement instead of HTMLScriptElement.

**Comment by irrationalRock** [comment]
Thanks, I appreciate the responses.
I moved `nonce` to `HTMLOrSVGElement` and `FocusOptions`.

**Comment by saschanaz** [comment]
HTMLOrSVGElement should also be added to `implements` list of HTMLElement.

FocusOptions should rather be added by https://html.spec.whatwg.org/multipage/interaction.html and also `focus()` signature should be overridden.

**Comment by irrationalRock** [comment]
Thanks for the feedback, I added `HTMLOrSVGElement` to `HTMLElement`'s list.
However, this causes HTMLElementMap to lose it's extends of `ElementEventMap`.
Also, I overridden the focus() signature.
@saschanaz I'm wondering if you can give more information on what you mean by FocusOptions should be added by https://html.spec.whatwg.org/multipage/interaction.html.

**Comment by saschanaz** [comment]
@irrationalRock `extends` shouldn't be overridden here, doing so will break EventMap generation which uses the field.

For FocusOptions, you can add this to inputfiles/idlSources.json:

```json
{
  "url": "https://html.spec.whatwg.org/multipage/interaction.html",
  "title": "HTML - User interaction"
}
```

...and then run `npm run fetch && npm run build && npm run baseline-accept`. It will automatically import FocusOptions from the spec document.

**Comment by irrationalRock** [comment]
I added `FocusOptions` with `inputfiles/idlSources.json`.

If `extends` shouldn't be used here, what would be the best place for it?

**Comment by saschanaz** [comment]
`implements` should be overridden but I'm afraid it won't work currently, as the merge function doesn't support arrays very well. Perhaps we may add `additional-implements` like the current `additional-signatures`.

**Comment by saschanaz** [comment]
Oops, I was wrong. Just adding implements: ["HTMLOrSVGElement"] will append to the current list.

File: inputfiles/overridingTypes.json [hunk 1]
```diff
@@ -897,6 +897,18 @@
                     }
                 }
             },
+            "HTMLElement": {
+                "methods": {
+                    "method": {
+                        "focus": {
+                            "name": "focus",
+                            "override-signatures": [
+                                "focus(options?: FocusOptions): void"
```
Comment (mhegazy) [inline inputfiles/overridingTypes.json]: so why is this needed?

Comment (saschanaz) [reply inputfiles/overridingTypes.json]: For https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus. (Separate from nonce, though.)
